### PR TITLE
Product Editor: show and locate help for the attribute terms component

### PR DESCRIPTION
**Pull request by retrofox**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

We've added a help section to the token fields component to help users create terms from the New Attribute Modal component.

Closes # .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to the new product editor
2. Go to the Variations tab
3. Add a new variation by clicking on the `Add options` button
4. Confirm the Terms input (FormTokenField) shows a help

Screenshot 2024-06-19 at 2 08 48 PM

5. Test the UI by adding any terms
6. Confirm it's okay: no weird overlaps, misalignments, etc

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/components/attribute-control/new-attribute-modal.scss [hunk 1]
```diff
@@ -90,14 +90,16 @@
 			position: absolute;
 			top: 36px;
 			z-index: 0;
-			display: none;
 		}
```
Comment (louwie17) [inline packages/js/product-editor/src/components/attribute-control/new-attribute-modal.scss:93]: We should remove all of the above styles, and that way we can also remove the styles added below.

**Review by louwie17 — CHANGES_REQUESTED** [review]
Having the help set to absolute means that as soon as the token field items move to a second line the help text disappears.
We should make it work as normal, where it is just rendered relative to the dropdown input field, see video:

https://github.com/woocommerce/woocommerce/assets/2240960/b8cefc89-dd15-4fa5-822d-4b9ecafed605

**Comment by retrofox** [comment]
> Having the help set to absolute means that as soon as the token field items move to a second line the help text disappears.

It's interesting I think it should move down according to the drop-down content. 
If the user could add terms means that they already know how to add them :-)

Moving the help to me feels weird. Also, it'll take more space in the UI. Something that inst good considering that the available space is not too big. 

Compartes with the atribuye input (ComboboxControl), the help disappears one the user creates/selects an attribute.

**Comment by louwie17** [comment]
> It's interesting I think it should move down according to the drop-down content. If the user could add terms means that they already know how to add them :-)
> 
> Moving the help to me feels weird. Also, it'll take more space in the UI. Something that inst good considering that the available space is not too big.
> 
> Compartes with the atribuye input (ComboboxControl), the help disappears one the user creates/selects an attribute.

Yes, in that case I think we should be consistent and match that of the `ComboboxControl` and make the help text disappear after a user has selected an attribute term.
This would be better than having it disappear do to a styling constraint with a double line.

File: packages/js/product-editor/src/components/attribute-control/new-attribute-modal.scss (hunk 1 above)
Comment (retrofox) [reply packages/js/product-editor/src/components/attribute-control/new-attribute-modal.scss:93]: No, we can't. There is an issue in the core that renders an empty element even when the `label` property is `null.` We can remove it once [these PR changes](https://github.com/WordPress/gutenberg/pull/61336) are available in WordPress core.

Comment (louwie17) [reply packages/js/product-editor/src/components/attribute-control/new-attribute-modal.scss:93]: Hmm is this also still relevant now that we are always showing it?

**Review by louwie17 — APPROVED** [review]
Left one remaining question around the help text staying around even when the selected items go to a new line, but otherwise these changes look good!
This can be a follow up.

File: packages/js/product-editor/src/components/attribute-control/new-attribute-modal.scss (hunk 1 above)
Comment (retrofox) [reply packages/js/product-editor/src/components/attribute-control/new-attribute-modal.scss:93]: image

If we remove those lines, the help moves down, something that we want?

**Pull request merged**